Sources/MapboxNavigation/Route.swift

```diff
@@ -15,19 +15,18 @@ extension Route {
      - returns: A polyline whose length is twice `distance` and whose centroid is located at the maneuver.
      */
     func polylineAroundManeuver(legIndex: Int, stepIndex: Int, distance: CLLocationDistance) -> LineString {
-        let precedingLegs = legs.prefix(upTo: legIndex)
-        let precedingLegCoordinates = precedingLegs.flatMap { $0.steps }.flatMap { $0.shape?.coordinates ?? [] }
-        
-        let precedingSteps = legs[legIndex].steps.prefix(upTo: stepIndex)
-        let precedingStepCoordinates = precedingSteps.compactMap { $0.shape?.coordinates }.reduce([], +)
-        let precedingPolyline = LineString((precedingLegCoordinates + precedingStepCoordinates).reversed())
+        var precedingCoordinates = [LocationCoordinate2D]()
+        if stepIndex > 0 {
```

Comment:
I'm not sure If I fully understand, so I'd like to clarify.

Previously we used all the previous legs and all the previous steps in the current leg. And now we'd like to use:
* **one** previous step from the current leg if there is such a step
*  up to **two** last steps from the last previous leg if there were no previous steps in the current leg

So I have a question, why do we use 1 last step in one case and 2 last steps in other case? Should this number be the same?

Reply:
Right now each `RouteLeg` has the last `RouteStep` as two duplicate coordinates as the arrival step. So if there were no previous steps in the current leg, which means that the previous step should be the last step for the previous `RouteLeg`. So we need the last `2` steps which contain the 0-length arrival step.